### PR TITLE
Add com.seafile.Client

### PR DESCRIPTION
**Pull request by eduardosm**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/haiwen/seafile-client/issues/1381
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.seafile.Client.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+app-id: com.seafile.Client
+runtime: org.kde.Platform
+runtime-version: '5.15'
```
Comment (hfiguiere) [inline com.seafile.Client.yaml:3]:
```suggestion
runtime-version: '5.15-21.08'
```

File: com.seafile.Client.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+rename-icon: seafile
+
+modules:
+  - name: vala-bootstrap
+    sources:
+      - type: archive
+        url: https://download.gnome.org/sources/vala/0.16/vala-0.16.1.tar.xz
+        sha256: 8be28ff6347feecb391220ceb4e2209d32b697278e87ed7fc6c9fb1e9345e0d8
+
+  - name: vala
```
Comment (hfiguiere) [inline com.seafile.Client.yaml:42]: are you sure you need vala?

**Comment by hfiguiere** [comment]
bot, build com.seafile.Client

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/com.seafile.Client.appdata.xml: FAILED:
• style-invalid         : Not enough <screenshot> tags, minimum is 1
```

Also I didn't see it submitted in the upstream issue linked.

File: com.seafile.Client.yaml (hunk 2 above)
Comment (eduardosm) [reply com.seafile.Client.yaml:42]: Yes, vala is needed for sure:
https://github.com/haiwen/seafile/blob/d01b8de6d378c69525e6ed57b84bd22579b96444/configure.ac#L28

File: com.seafile.Client.yaml (hunk 1 above)
Comment (eduardosm) [reply com.seafile.Client.yaml:3]: Fails to build on `5.15-21.08`, it looks like there is a regression and some path is not set correctly:

```
----------------------------------------------------------------------
 /usr/bin/mkdir -p '/app/include'
 /usr/bin/install -c -m 644 searpc-client.h searpc-server.h searpc-utils.h searpc.h searpc-named-pipe-transport.h '/app/include'
make[2]: Leaving directory '/run/build/libsearpc/lib'
make[1]: Leaving directory '/run/build/libsearpc/lib'
Making install in pysearpc
make[1]: Entering directory '/run/build/libsearpc/pysearpc'
make[2]: Entering directory '/run/build/libsearpc/pysearpc'
make[2]: Nothing to be done for 'install-exec-am'.
 /usr/bin/mkdir -p '/usr/lib/python3.9/site-packages/pysearpc'
/usr/bin/mkdir: cannot create directory ‘/usr/lib/python3.9/site-packages/pysearpc’: Read-only file system
make[2]: *** [Makefile:338: install-pysearpcPYTHON] Error 1
make[2]: Leaving directory '/run/build/libsearpc/pysearpc'
make[1]: *** [Makefile:431: install-am] Error 2
make[1]: Leaving directory '/run/build/libsearpc/pysearpc'
make: *** [Makefile:476: install-recursive] Error 1
Error: module libsearpc: El proceso hijo terminó con el código 2
```

**Comment by eduardosm** [comment]
> ```
> builddir/files/share/appdata/com.seafile.Client.appdata.xml: FAILED:
> • style-invalid         : Not enough <screenshot> tags, minimum is 1
> ```

Added

> Also I didn't see it submitted in the upstream issue linked.

Does it need to be submitted upstream? I've seen many applications where it hasn't.

File: com.seafile.Client.yaml (hunk 1 above)
Comment (eduardosm) [reply com.seafile.Client.yaml:3]: Fixed by manually adding `--with-python_prefix=/app`

File: com.seafile.Client.yaml (hunk 2 above)
Comment (hfiguiere) [reply com.seafile.Client.yaml:42]: ok. I thought that building from tarball didn't require Vala.

**Comment by hfiguiere** [comment]
> > ```
> > builddir/files/share/appdata/com.seafile.Client.appdata.xml: FAILED:
> > • style-invalid         : Not enough <screenshot> tags, minimum is 1
> > ```
> 
> Added
> 
> > Also I didn't see it submitted in the upstream issue linked.
> 
> Does it need to be submitted upstream? I've seen many applications where it hasn't.

Yes it does. But if upstream doesn't accept it, ignore the request, etc.

Note: this is not flatpak specific either.

**Comment by hfiguiere** [comment]
bot, build com.seafile.Client

**Comment by eduardosm** [comment]
I opened a PR to include appstream metadata upstream: https://github.com/haiwen/seafile-client/pull/1388

**Comment by nedrichards** [comment]
bot, build com.seafile.Client

File: com.seafile.Client.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
+app-id: com.seafile.Client
+runtime: org.kde.Platform
+runtime-version: '5.15-21.08'
+sdk: org.kde.Sdk
+command: seafile-applet
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --share=network
+  - --persist=.ccnet
```
Comment (nedrichards) [inline com.seafile.Client.yaml:10]: Qt complains about not being able to get GPU acceleration, adding `--device=dri` fixes that.

**Comment by nedrichards** [comment]
Other than --device=dri this lgtm

File: com.seafile.Client.yaml (hunk 3 above)
Comment (eduardosm) [reply com.seafile.Client.yaml:11]: Fixed

**Comment by barthalion** [comment]
bot, build com.seafile.Client

File: com.seafile.Client.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline com.seafile.Client.metainfo.xml:1]: Have you submitted this file upstream?

File: com.seafile.Client.yaml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+runtime-version: '5.15-21.08'
+sdk: org.kde.Sdk
+command: seafile-applet
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --share=network
+  - --device=dri
+  - --persist=.ccnet
+  - --filesystem=home/Seafile:create # Where user libraries are stored by default
```
Comment (barthalion) [inline com.seafile.Client.yaml:12]: Does that actually work? I've seen `~/Seafile` in the past, not `home`.

Comment (barthalion) [inline com.seafile.Client.yaml:12]: Does that actually work? I've seen `~/Seafile` in the past, not `home`.

File: com.seafile.Client.metainfo.xml (hunk 4 above)
Comment (eduardosm) [reply com.seafile.Client.metainfo.xml:1]: https://github.com/haiwen/seafile-client/pull/1388

File: com.seafile.Client.yaml (hunk 5 above)
Comment (eduardosm) [reply com.seafile.Client.yaml:12]: I'm not sure what you mean. `--filesystem=home/Seafile` means that the application has access to `~/Seafile`, doesn't it?

Comment (nedrichards) [reply com.seafile.Client.yaml:12]: @barthalion I now have a Seafile folder in my home directory so it does seem to!

**Pull request closed without merging**